board/common/overlay/sbin/fwupdate

```diff
@@ -111,7 +115,7 @@ function do_download() {
     version=$1
 
     if ! [[ "$url" == http* ]]; then  # a version was given
-        url=$(show_versions true | grep "^$1" | cut -d ' ' -f 2)
+        url=$(show_versions true | sed -rn '/^'"$1"' http.*\.img\.[a-z]+$/ {;/.*\.xz/ !{;h}; /.*\.xz/ {;s/^'"$1"' (.*)/\1/p;q;};}; $ {;x;s/^'"$1"' (.*)/\1/p;}')
```

Comment:
We might want to support other compressed images as well (such as `.zip`, since Windows users still prefer `.zip` files). Wouldn't it be a good idea to have and use a list of supported archive extensions, somewhere in a variable?

Reply:
Do we want to support ZIP format for __remote firmware upgrade__? The purpose of supporting xz is to reduce the firmware file size so it takes less time to download, thus speeding up the upgrade process and reduces the chance of failure. ZIP format would give us the same file size as gz format, so won't give us any benefit in __remote firmware upgrade__. I can see the reason to have all three formats available.

We can definitely support ZIP format in the future, and add provision for it now.
For now, what about this line that allows us to easily add ZIP format later:
`url=$(show_versions true | sed -rn '/^'"$version"' http.*\.img\.[a-z]+$/ {; /.*\.xz$/ {;s/^'"$version"' (.*)/\1/ p;q;}; /.*\.gz$/ {;h;b;};}; $ {;x;s/^'"$version"' (.*)/\1/ p;}')`

In the future, we can change to this line that supports ZIP format as well:
`url=$(show_versions true | sed -rn '/^'"$version"' http.*\.img\.[a-z]+$/ {; /.*\.xz$/ {;s/^'"$version"' (.*)/\1/ p;q;}; /.*\.gz$/ {;h;b;}; /.*\.zip$/ {;x;/.*\.gz$/ {;x;b;};b;};}; $ {;x;s/^'"$version"' (.*)/\1/ p;}')`

Explanation:
```
sed -rn
'/^'"$version"' http.*\.img\.[a-z]+$/ {
    /.*\.xz$/ {
        # Found xz image, print url and quit
        s/^'"$version"' (.*)/\1/ p
        q
    }
    /.*\.gz$/ {
        # Found gz image, copy to hold buffer and branch to end of script
        h
        b
    }
    /.*\.zip$/ {
        # Found zip image, have we already found gz image?
        x
        /.*\.gz$/ {
            # We already found gz image, do nothing, branch to end of script
            x
            b
        }
        # zip image is already in hold buffer, branch to end of script
        b
    }
}
$ {
    # We have reached the end of input stream, print hold buffer
    x
    s/^'"$version"' (.*)/\1/ p
}'
```

Essentially, the one liner looks for xz image first, then gz format 2nd. In the future, it can look for zip format 3rd.